sympy/simplify/simplify.py

```diff
@@ -522,6 +522,10 @@ def simplify(expr, ratio=1.7, measure=count_ops, fu=False):
         return expr
 
     if not isinstance(expr, (Add, Mul, Pow, ExpBase)):
+        if isinstance(expr, Function) and hasattr(expr, "inverse"):
+            if isinstance(expr.args[0], expr.inverse()):
```

Comment:
It has to do with functions of multiple variables. However, I just looked in the solvers and it doesn't look like they are handling it either. I don't think inverse with argindex other than 1 is used anywhere in SymPy at this time.